lib/distribute_reads/appropriate_pool.rb

```diff
@@ -5,12 +5,14 @@ def _appropriate_pool(*args)
         if Thread.current[:distribute_reads][:replica]
           if @slave_pool.completely_blacklisted?
             raise DistributeReads::NoReplicasAvailable, "No replicas available" if Thread.current[:distribute_reads][:failover] == false
+            warn "No replicas available. Falling back to master pool." if Thread.current[:distribute_reads][:log_failover]
             @master_pool
           else
             @slave_pool
           end
         elsif Thread.current[:distribute_reads][:primary] || needs_master?(*args) || (blacklisted = @slave_pool.completely_blacklisted?)
           raise DistributeReads::NoReplicasAvailable, "No replicas available" if blacklisted && Thread.current[:distribute_reads][:failover] == false
+          warn "No replicas available. Falling back to master pool." if blacklisted && Thread.current[:distribute_reads][:log_failover]
```

Comment:
Would be good to prefix message with “[distribute_reads] ” so users know what’s generating it.